packages/sdk/src/browser-wallets/magiceden/index.ts

```diff
@@ -36,17 +45,8 @@ function isInstalled(): boolean {
   );
 }
 
-async function getMagicEdenWalletProvider(): Promise<BitcoinProvider> {
-  if (!isInstalled()) {
-    throw new BrowserWalletNotInstalledError(
-      "Magic Eden not installed or set as prioritised wallet.",
-    );
-  }
-
-  return window.BitcoinProvider!;
-}
-
 async function getAddresses(
+  getMeProvider: () => Promise<BitcoinProvider>,
```

Comment:
we've modified the function signature, how will a user get the magic eden provider? can we obtain this internally?

Reply:
i think it's possible
```
import { getWallets } from "@wallet-standard/core";
  const meWallet = getWallets()
    .get()
    .find((wallet) => wallet.name === "Magic Eden");
  const magicEdenWalletProvider = (meWallet as MagicEdenWallet)?.features[
    "sats-connect:"
  ]!.provider;

```
the result should be the same instead of passing the provider from the web